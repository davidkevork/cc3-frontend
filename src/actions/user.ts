import 'whatwg-fetch';
import { action } from 'typesafe-actions';
import { API_URL } from '../constants/const';

export const UserActionTypes = {
  USER_FROM_TOKEN: 'USER_FROM_TOKEN',
  USER_FROM_TOKEN_SUCCESS: 'USER_FROM_TOKEN_SUCCESS',
  USER_FROM_TOKEN_FAILURE: 'USER_FROM_TOKEN_FAILURE',

  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  LOGOUT_FAILURE: 'LOGOUT_FAILURE',

  RESET_USER_TOKEN: 'RESET_USER_TOKEN',
  SET_USER_LOADING: 'SET_USER_LOADING',
  SET_USER_ERROR: 'SET_USER_ERROR',
  CLEAR_USER_ERROR: 'CLEAR_USER_ERROR',
}

/** User from token */
export const userFromToken = (
  tokenFromStorage: string,
  userId: string,
): any => action(UserActionTypes.USER_FROM_TOKEN, fetch(`${API_URL}users/${userId}`, {
  method: 'GET',
  headers: new Headers({
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${tokenFromStorage}`,
  }),
}).then(response => response.json()));

export const userFromTokenFailure = (
  error: Error,
) => action(UserActionTypes.USER_FROM_TOKEN_FAILURE, error);

export const userFromTokenSuccess = (
  user: object,
) => action(UserActionTypes.USER_FROM_TOKEN_SUCCESS, user);

/** Logout */
export const logoutSuccess = () => action(UserActionTypes.LOGOUT_SUCCESS, null);

export const logoutFailure = (
  error: Error,
) => action(UserActionTypes.LOGOUT_FAILURE, error);

/** Common actions */
export const resetUserToken = (
) => action(UserActionTypes.RESET_USER_TOKEN);

export const setUserLoading = (
) => action(UserActionTypes.SET_USER_LOADING);

export const setUserError = (
  error: Error,
) => action(UserActionTypes.SET_USER_ERROR, error);

export const clearUserError = (
) => action(UserActionTypes.CLEAR_USER_ERROR);
