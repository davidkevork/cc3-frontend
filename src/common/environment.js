export const environment = {
  host: 'http://ec2-13-236-4-20.ap-southeast-2.compute.amazonaws.com:8080'
};