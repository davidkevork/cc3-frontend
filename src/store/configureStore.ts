import { Store, createStore, applyMiddleware } from 'redux';
import promise from 'redux-promise';
import rootReducer, { RootState } from '../reducers';

const enhancer = applyMiddleware(
  promise,
);

export const configureStore = (initialState?: any): Store<RootState> => {
  return createStore(rootReducer, initialState, enhancer);
};
