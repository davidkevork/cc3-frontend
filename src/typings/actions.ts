export interface valueAction<T> {
  type: string;
  payload: T;
}