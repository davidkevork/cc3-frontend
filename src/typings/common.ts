export type JsonObject = Record<any, any>;