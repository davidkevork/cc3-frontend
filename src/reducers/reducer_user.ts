import { Reducer } from 'redux';
import { UserActionTypes } from '../actions/user';

export interface IUserReducerState {
  readonly jwt: string | undefined;
  readonly userId: string | undefined;
  readonly email: string;
  readonly status: string;
  readonly error: Error | undefined;
  readonly loading: boolean;
}

export const INITIAL_STATE: IUserReducerState = {
  jwt: undefined,
  userId: undefined,
  email: '',
  status: '',
  error: undefined,
  loading: false,
};

/* tslint:disable:variable-name */
export const UserReducer: Reducer<IUserReducerState> = (
  state = INITIAL_STATE,
  action,
) => {
  const {
    jwt, userId, email,
  }: {
    jwt: string;
    userId: string;
    email: string;
  } = (action.payload === undefined
    || action.payload === null
    || typeof action.payload !== 'object') ? {} : action.payload;

  switch (action.type) {
    case UserActionTypes.USER_FROM_TOKEN:
      return {
        ...state,
        jwt: undefined,
        userId: undefined,
        status: 'STORAGE',
        error: undefined,
        loading: true,
      };
    case UserActionTypes.USER_FROM_TOKEN_SUCCESS:
      return {
        ...state,
        jwt,
        userId,
        email,
        status: 'AUTHENTICATED',
        error: undefined,
        loading: false,
      };
    case UserActionTypes.USER_FROM_TOKEN_FAILURE:
      return {
        ...state,
        jwt: undefined,
        userId: undefined,
        status: 'STORAGE',
        error: action.payload,
        loading: false,
      };

    case UserActionTypes.LOGOUT_SUCCESS:
      return {
        ...state,
        ...INITIAL_STATE,
        status: UserActionTypes.LOGOUT_SUCCESS,
        error: undefined,
        loading: false,
      };
    case UserActionTypes.LOGOUT_FAILURE:
      return {
        ...state,
        status: UserActionTypes.LOGOUT_FAILURE,
        error: action.payload,
        loading: false,
      };

    case UserActionTypes.RESET_USER_TOKEN:
      return {
        ...state,
        jwt: undefined,
        userId: undefined,
        status: 'STORAGE',
        error: undefined,
        loading: false,
      };
    case UserActionTypes.SET_USER_LOADING:
      return {
        ...state,
        loading: true,
        status: UserActionTypes.SET_USER_LOADING,
      };
    case UserActionTypes.SET_USER_ERROR:
      return {
        ...state,
        status: UserActionTypes.SET_USER_ERROR,
        error: action.payload,
      };
    case UserActionTypes.CLEAR_USER_ERROR:
      return {
        ...state,
        status: UserActionTypes.CLEAR_USER_ERROR,
        error: action.payload,
      };
    default:
      return state;
  }
};
