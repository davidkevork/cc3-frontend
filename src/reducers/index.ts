import { combineReducers } from 'redux';
import { IUserReducerState, UserReducer } from './reducer_user';

export interface RootState {
  user: IUserReducerState;
}

const rootReducer = combineReducers({
  user: UserReducer,
});

export default rootReducer;
