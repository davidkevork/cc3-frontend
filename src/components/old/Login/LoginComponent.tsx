import * as React from 'react';
import { NavLink } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import intl from 'react-intl-universal';
import { hot } from 'react-hot-loader';
import { enUS } from '../../locales/en-US';
import { IUserBusinessReducerState } from '../../reducers/reducer_user_business';
import { ILoginReducerState } from '../../reducers/reducer_login';
import { errorType } from '../../../typings/common/actions';
import { emailSchema, passwordSchema } from '../../schemas/inputSchema';
import IndexNav from '../../IndexNav/IndexNav';
import AuthTabs from '../../AuhTabs/AuthTabs';
import './Login.scss';
import { bg_card_auth, bg_business_auth } from '../../constants/images';

const isUser = process.env.ROLE === 'user';

export interface ILoginComponentProps {
  handleChange: (type: string, value: string, error: errorType | undefined) => void;
  redirect: (to: string) => void;
  clearFields: () => void;
  loadProfile: (props: IUserBusinessReducerState) => void;
  submit: (email: string, password: string, callback: () => void) => any;
  userBusiness: IUserBusinessReducerState;
  login: ILoginReducerState;
}

class LoginComponent extends React.Component<ILoginComponentProps, {}> {
  constructor(props: ILoginComponentProps) {
    super(props);
  }
  public componentWillReceiveProps(nextProps: ILoginComponentProps) {
    if (nextProps.userBusiness.status === 'authenticated'
    && nextProps.userBusiness.userId
    && !nextProps.userBusiness.error) {
      this.props.redirect('/cards');
    }
  }
  public componentWillUnmount() {
    this.props.clearFields();
  }
  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.props.handleChange(event.target.id, event.target.value, this.props.login.error);
  }
  private submit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (!this.props.userBusiness.loading) {
      this.props.submit(this.props.login.email, this.props.login.password, () => {
        this.props.loadProfile(this.props.userBusiness);
      });
    }
  }
  public render() {
    return (
      <Grid className="page page-login page-auth">
        <IndexNav />
        <Grid
          container={true}
          spacing={16}
          alignItems="center"
          className="page-content"
          direction="column"
        >
          <h2 className="title">
            {isUser ?
              intl.get('WELCOME_CARD').d(enUS.WELCOME_CARD)
              :
              intl.get('WELCOME_BUSINESS').d(enUS.WELCOME_BUSINESS)
            }
          </h2>
          {/* <h2 className="title text-center">{intl.get('LOGIN').d(enUS.LOGIN)}</h2> */}
          <Grid container={true} className="page-wrapper">
            <Grid item={true} className="page-image">
              <img src={isUser ? bg_card_auth : bg_business_auth} alt="Login Image"/>
            </Grid>
            <Grid item={true} className="page-form">
              <AuthTabs value="login" />
              <form method="post" onSubmit={this.submit}>
                <Grid container={true} className="form-inputs">
                  <TextField
                    id="email"
                    label={intl.get('EMAIL_ADDRESS').d(enUS.EMAIL_ADDRESS)}
                    margin="normal"
                    type="email"
                    fullWidth={true}
                    inputProps={emailSchema}
                    onChange={this.handleChange}
                    required={emailSchema.required}
                  />
                  <TextField
                    id="password"
                    label={intl.get('PASSWORD').d(enUS.PASSWORD)}
                    margin="normal"
                    type="password"
                    fullWidth={true}
                    inputProps={passwordSchema}
                    onChange={this.handleChange}
                    required={passwordSchema.required}
                  />
                </Grid>
                <Grid
                  container={true}
                  alignContent="center"
                  alignItems="center"
                  justify="space-between"
                  className="form-buttons"
                  spacing={16}
                >
                  <NavLink className="password-reset" to="/reset">
                    {intl.get('FORGOT_PASSWORD').d(enUS.FORGOT_PASSWORD)}
                  </NavLink>
                  <Button variant="contained" color="primary" type="submit">
                    {intl.get('LOGIN').d(enUS.LOGIN)}
                  </Button>
                </Grid>
              </form>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default hot(module)(LoginComponent);
