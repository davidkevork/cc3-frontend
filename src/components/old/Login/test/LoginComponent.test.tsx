import * as React from 'react';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import promise from 'redux-promise';
import LoginComponent from '../LoginComponent';
import { UserBusinessReducer } from '../../../reducers/reducer_user_business';
import { LoginReducer } from '../../../reducers/reducer_login';
import { errorType } from '../../../../typings/common/actions';

const mockStore = configureStore([promise]);

describe('Login Component', () => {
  it('should render correctly', () => {
    const store = mockStore({
      login: LoginReducer,
      userBusiness: UserBusinessReducer,
    });
    let LoginComponentRendered = enzyme.shallow(
      <LoginComponent
        handleChange={(type: string, value: string, error: errorType | undefined) => {}}
        redirect={(to: string) => {}}
        clearFields={() => {}}
        submit={(email: string, password: string) => {return true}}
        userBusiness={store.userBusiness}
        login={store.login}
      />
    );
    expect(toJson(LoginComponentRendered)).toMatchSnapshot();
  });