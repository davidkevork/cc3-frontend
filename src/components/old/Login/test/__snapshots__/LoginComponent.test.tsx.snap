// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render correctly 1`] = `
<WithStyles(Grid)
  className="page page-login page-auth"
>
  <Connect(IndexNav) />
  <WithStyles(Grid)
    alignItems="center"
    className="page-content"
    container={true}
    direction="column"
    spacing={16}
  >
    <h2
      className="title"
    >
      Welcome to SmartCash for Business
    </h2>
    <WithStyles(Grid)
      className="page-wrapper"
      container={true}
    >
      <WithStyles(Grid)
        className="page-image"
        item={true}
      >
        <img
          alt="Login Image"
          src="test-file-stub"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="page-form"
        item={true}
      >
        <AuthTabs
          value="login"
        />
        <form
          method="post"
          onSubmit={[Function]}
        >
          <WithStyles(Grid)
            className="form-inputs"
            container={true}
          >
            <TextField
              fullWidth={true}
              id="email"
              inputProps={
                Object {
                  "required": true,
                }
              }
              label="Email address"
              margin="normal"
              onChange={[Function]}
              required={true}
              select={false}
              type="email"
              variant="standard"
            />
            <TextField
              fullWidth={true}
              id="password"
              inputProps={
                Object {
                  "maxLength": 100,
                  "minLength": 25,
                  "required": true,
                }
              }
              label="Password"
              margin="normal"
              onChange={[Function]}
              required={true}
              select={false}
              type="password"
              variant="standard"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            alignContent="center"
            alignItems="center"
            className="form-buttons"
            container={true}
            justify="space-between"
            spacing={16}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="password-reset"
              to="/reset"
            >
              Forgot Password?
            </NavLink>
            <WithStyles(Button)
              color="primary"
              type="submit"
              variant="contained"
            >
              Login
            </WithStyles(Button)>
          </WithStyles(Grid)>
        </form>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
